dask_match/expr.py

```diff
@@ -1003,15 +962,17 @@ def _name(self):
     def _divisions(self):
         return self.exprs[0]._divisions()
 
-    def dependencies(self):
-        return self.operands[1:]
-
-    def _blockwise_layer(self):
-        block = {self._name: self.exprs[0]._name}
+    def _task(self, index):
+        graph = {self._name: (self.exprs[0]._name, index)}
```

Comment:
Are you worried about re-constructing this graph in a loop for every task? One of the advantages of `SubgraphCallable` is that we only generate it once and then *use* it in every task.

Reply:
I am mildly concerned about that cost.  I'm far more concerned about the complexity of abstractions that was building up though.  It was getting hard for me to follow, which probably means that it'll be hard for other developers to work on as well.

At the end of the day we're going to be linear in the size of that graph anyway.  For example we're going to need to serialize each subgraphcallable separately to send to the workers.  So we're going to be bound close to some cost similar to this no matter what.